homeassistant/components/sensor/xiaomi_miio.py

```diff
@@ -0,0 +1,209 @@
+"""
+Support for Xiaomi Mi Air Quality Monitor (PM2.5).
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/sensor.xiaomi_miio/
+"""
+import asyncio
+from functools import partial
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import (PLATFORM_SCHEMA, DOMAIN)
+from homeassistant.const import (CONF_NAME, CONF_HOST, CONF_TOKEN,
+                                 SERVICE_TURN_ON, SERVICE_TURN_OFF,
+                                 ATTR_ENTITY_ID, )
+from homeassistant.exceptions import PlatformNotReady
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Xiaomi Miio Sensor'
+DATA_KEY = 'sensor.xiaomi_miio'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_TOKEN): vol.All(cv.string, vol.Length(min=32, max=32)),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+REQUIREMENTS = ['python-miio==0.3.7']
+
+ATTR_POWER = 'power'
+ATTR_CHARGING = 'charging'
+ATTR_BATTERY_LEVEL = 'battery_level'
+ATTR_TIME_STATE = 'time_state'
+ATTR_MODEL = 'model'
+
+SUCCESS = ['ok']
+
+SERVICE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+SERVICE_TO_METHOD = {
+    SERVICE_TURN_ON: {'method': 'async_turn_on'},
```

Comment:
Should I use `xiaomi_miio_turn_on` instead of `turn_on` here?

Reply:
For that I don't know an answer. I suppose one should generally use the already existing `turn_on`/`turn_off` calls, but for sensors there exists no such? Anyway, I suppose this should be `turn_on` for consistency.